### PR TITLE
Add support for pipeline

### PR DESCRIPTION
**Pull request by lli-parasoft**

### CODE REVIEW

File: src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Set;
+
+public class ParasoftCoverageStep extends Step implements Serializable {
+    private static final long serialVersionUID = -2235239576082380147L;
+    private String pattern = StringUtils.EMPTY;
+    private static final String DEFAULT_PATTERN = "**/coverage.xml";
```
Comment (awang-parasoft) [inline src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java:29]: Maybe we can change the access modifier of these constants in coverage recorder to package-private(no modifier) and use them in this class, then we don't need to define them here.

File: src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+            tool.setParser(CoverageTool.Parser.COBERTURA);
+            tool.setPattern(coverageResult.getCoberturaPattern());
+            CoverageStep coverageStep = new CoverageStep();
+            coverageStep.start(stepContext).getContext().get(TaskListener.class);
+            coverageStep.setTools(List.of(tool));
+            coverageStep.setId(PARASOFT_COVERAGE_ID);
+            coverageStep.setName(PARASOFT_COVERAGE_NAME);
+            StepExecution execution = coverageStep.start(stepContext);
+            execution.start();
+            return UNUSED;
```
Comment (awang-parasoft) [inline src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java:79]: To keep consistency with free-style job, we also need to delete temporary coverage directories after the coverage is published to Jenkins.

File: src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+            CoverageTool tool = new CoverageTool();
+            var parasoftCoverageRecorder = new ParasoftCoverageRecorder();
+            parasoftCoverageRecorder.setPattern(step.getPattern());
+            LogHandler logHandler = new LogHandler(getTaskListener(), PARASOFT_COVERAGE_NAME);
+            ParasoftCoverageRecorder.CoverageConversionResult coverageResult = parasoftCoverageRecorder.performCoverageReportConversion(
+                    getRun(), getWorkspace(), logHandler, new RunResultHandler(getRun()));
+            tool.setParser(CoverageTool.Parser.COBERTURA);
+            tool.setPattern(coverageResult.getCoberturaPattern());
+            CoverageStep coverageStep = new CoverageStep();
+            coverageStep.start(stepContext).getContext().get(TaskListener.class);
```
Comment (awang-parasoft) [inline src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java:73]: Is this line necessary?

File: src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+        @Override
+        public String getFunctionName() {
+            return "parasoftCoverage";
+        }
+
+        @NonNull
+        @Override
+        public String getDisplayName() {
+            return "Record Parasoft code coverage results";
```
Comment (awang-parasoft) [inline src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java:99]: We need to use localized message here. See the same method in coverage recorder.

**Review by awang-parasoft — CHANGES_REQUESTED** [review]

**Review by awang-parasoft — COMMENTED** [review]
OK

File: src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+            FilePath workspace = getWorkspace();
+            TaskListener taskListener = getTaskListener();
+            RunResultHandler runResultHandler = new RunResultHandler(run);
+            var parasoftCoverageRecorder = new ParasoftCoverageRecorder();
+            parasoftCoverageRecorder.setPattern(step.getPattern());
+            LogHandler logHandler = new LogHandler(taskListener, PARASOFT_COVERAGE_NAME);
+            ParasoftCoverageRecorder.CoverageConversionResult coverageResult = parasoftCoverageRecorder.performCoverageReportConversion(
+                    run, workspace, logHandler, runResultHandler);
+            CoverageRecorder recorder = setUpCoverageRecorder(coverageResult.getCoberturaPattern());
+            Method performMethod =
```
Comment (yshi-parasoft) [inline src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java:70]: Please add a comment to explain the reason for using Reflection here.

File: src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                    ReflectionUtils.findMethod(CoverageRecorder.class, "perform", Run.class, FilePath.class,
```
Comment (yshi-parasoft) [inline src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java:71]: Please add "//$NON-NLS-1$" for all places using strings but have no localization

**Pull request merged**